docs/source/en/generation_strategies.mdx

```diff
@@ -0,0 +1,304 @@
+text outputs for different tasks:
+* [Text summarization](./tasks/summarization#inference)
+* [Image captioning](./model_doc/git#transformers.GitForCausalLM.forward.example)
+* [Audio transcription](./model_doc/whisper#transformers.WhisperForConditionalGeneration.forward.example)
+
+Note that the inputs to the generate method depend on the model's modality. They can be `input_ids` for text, `input_values`
+or `input_features` for speech and `pixel_values` for computer vision. When in doubt, refer to the model's documentation.
+
+The process of selecting output tokens to generate text is known as decoding, and you can customize the decoding strategy
+that the `generate()` method will use. Modifying a decoding strategy does not change the values of any trainable parameters.
+However, it can have a noticeable impact on the quality of the generated output. It can help reduce repetition in the text
+and make it more coherent.
+
+This guide describes:
+* default generation configuration
+* common decoding strategies and their main parameters
+* saving and sharing custom generation configurations with your fine-tuned model on 🤗 Hub
+
+## Default text generation configuration
+
+A decoding strategy for a model is defined in its generation configuration. When using pre-trained models for inference
+within a [`pipeline`], the models call the `PreTrainedModel.generate()` method that applies a default generation
+configuration under the hood. The default configuration is also used when no custom configuration has been saved with
+the model.
+
+When you load a model explicitly, you can inspect the generation configuration that comes with it through
+ `model.generation_config`:
+
+```python
+>>> from transformers import AutoModelForCausalLM
+
+>>> model = AutoModelForCausalLM.from_pretrained("distilgpt2")
+>>> model.generation_config
+GenerationConfig {
+    "_from_model_config": true,
+    "bos_token_id": 50256,
+    "eos_token_id": 50256,
+    "transformers_version": "4.26.0.dev0"
+}
+```
+
+Printing out the `model.generation_config` reveals only the values that are different from the default generation
+configuration, and does not list any of the default values.
+
+The default generation configuration limits the size of the output combined with the input prompt to a maximum of 20
+tokens to avoid running into resource limitations. The default decoding strategy is greedy search, which is the simplest decoding strategy that picks a token with the highest probability as the next token. For many tasks
+and small output sizes this works well. However, when used to generate longer outputs, greedy search can start
+producing highly repetitive results.
+
+## Customize text generation
+
+You can override any `generation_config` by passing the parameters and their values directly to the [`generate`] method:
+
+```python
+>>> my_model.generate(inputs, num_beams=4, do_sample=True)
```

Comment:
```suggestion
>>> my_model.generate(**inputs, num_beams=4, do_sample=True)
```

In all examples, I'm going to ask to change the first argument to `generate` from `inputs` (a single tensor) to `**inputs` (an unpacked dictionary of tensors), or similar.

This is because the tokenizer/preprocessor may prepare more than one input to the model -- for instance, in NLP models, the tokenizer produces at least the input tokens and the attention mask. While `generate` often works well without these secondary inputs, it has to make assumptions about them (and fails sometimes). Since we are writing these amazing docs from scratch, we can take the opportunity to guide the users towards a better use of generate 🤗 

I've double-checked the stand-alone examples and confirmed that they work after the change.

Reply:
Thanks a ton for the detailed explanation! I appreciate it!